File: Assignment3638/program12.c
//write a recursive program which accept number from user and return Largest digit.
// Input : 87983
// OutPut : 9

#include<stdio.h>

int Maximum(int iNo)
{
  static int iMax = 0;
  static int iDigit = 0;

  if(iNo != 0)
  {
    iDigit = iNo%10;
    if(iMax < iDigit)
    {
      iMax = iDigit;
    }
    iNo = iNo/10;
    
    Maximum(iNo);
  }
  return iMax;
}

int main()
{
  int iValue = 0;
  int iRet = 0;

  printf("Enter the Number\n");
  scanf("%d",&iValue);

    iRet = Maximum(iValue);
    printf("Largest Number : %d\n",iRet);
    return 0;
}

File: Assignment47&50/program12.java
//write java program which accept number from user and return the count of odd digits

import java.lang.*;
import java.util.*;

class Digit
{
    public int Countodd(int iNo)
    {
        int idigit = 0, iCnt = 0;
        int i = 0;  

        if(iNo == -iNo)
        {
            iNo = -iNo;
        }
        while(iNo != 0)
    {
        idigit = iNo % 10;
        if(idigit % 2 != 0)
        {
            iCnt++;
        }
        iNo = iNo/10;
    }
        return iCnt;
    }
   
}


class program12
{
    public static void main(String arg[])
    {
        
        int iNo = 0;
        
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the Number..");
        iNo = sobj.nextInt();

        Digit dobj = new Digit();
       int iRet = dobj.Countodd(iNo);
        System.out.println("count of odd Number : "+iRet);
        
    }
}